design/design_resort.md

```diff
@@ -50,7 +50,15 @@ All the images are from [Berrycamp](https://berrycamp.github.io). Check out the
 
 # Chapter Arc
 
-It should be said that first 4-6 levels are single screen ones and scrolling is gradually added later in the chapter. The most scrolling would be found in mess levels and roof levels.
+It should be said that first 4-6 levels are single screen ones and scrolling is gradually introduced later in the chapter. The most scrolling would be found in mess levels and roof levels.
+
+# Oshiro boss fight
+
+Because the fight takes place lineraly, it wouldn't be a challenge to implement. The boss is to be directly adapted for pico-8 from the Celeste 2018 version. The rooms for Oshiro should mostly be the same. Some rooms can be removed upon discussion over it and some new fan-made rooms can be added.
+
+Due to the already substantial amount of rooms for the resort chapter, the Oshiro boss fight should be limited to 6 long scrolling rooms. 2 of them being inside the hotel, while the other four being on the roof. At the end the cutscene shall play as in the 2018 version.
+
+There shall not be a need to keep track of Oshiro health as the fight is only over when madeline reaches the last room.
```

Comment:
This line isn't needed, it's not something that exists in the original game either